website/i18n/zh/docusaurus-plugin-content-blog/2022/01/11/building-apisix-in-ubuntu-for-arm.md

```diff
@@ -0,0 +1,247 @@
+---
+title: "在 arm 版 ubuntu 构建 APISIX"
+authors:
+  - name: "郭奇"
+    title: "Author"
+    url: "https://github.com/guoqqqi"
+    image_url: "https://avatars.githubusercontent.com/u/72343596?v=4"
+keywords: 
+- Apache APISIX
+- arm
+- ubuntu
+- m1
+description: 本文为大家描述了在 M1 中启用 arm 版 ubuntu 通过源码构建 Apache APISIX 的详细操作步骤，通过阅读本文，大家对于在 arm ubuntu 上构建 Apache APISIX 更加快捷。
+tags: [Technology]
+---
+
+> 本篇文章记录了通过源码在 arm 版 ubuntu 中构建 Apache APISIX 的详细踩坑记录，ubuntu 环境使用的是 [https://multipass.run/](https://multipass.run/) 中的 Multipass 。因为它可以在 M1 上使用～
+
+<!--truncate-->
+
+## 克隆源码
+
+首先根据 [官方文档](https://apisix.apache.org/zh/docs/apisix/how-to-build#%E9%80%9A%E8%BF%87%E6%BA%90%E7%A0%81%E5%8C%85%E5%AE%89%E8%A3%85) 安装 APISIX 源码仓库。并且进入项目文件。
+
+```shell
+cd apisix-2.11.0
+```
+
+## 安装项目依赖
+
+1. 现在我们可以通过脚本一键安装项目所需要的依赖拉，在根目录下运行下面命令：
+
+```shell
+bash utils/install-dependencies.sh
+```
+
+![1.png](https://guoqi-test-1307026204.file.myqcloud.com/202108/1.png)
+
+不出所料，没有成功跑完。通过错误提示我们知道，这里是未能成功安装 `OpenResty`。原因是默认没有 `arm 64` 平台的源。
+
+2. 下面我们手动安装下 OpenResty：参考 [https://openresty.org/cn/linux-packages.html#ubuntu](https://openresty.org/cn/linux-packages.html#ubuntu)
+
+- 步骤一：安装导入 GPG 公钥时所需的几个依赖包（整个安装过程完成后可以随时删除它们）：
+
+```shell
+sudo apt-get -y install --no-install-recommends wget gnupg ca-certificates
+```
+
+- 步骤二：导入我们的 GPG 密钥：
+
+```shell
+wget -O - https://openresty.org/package/pubkey.gpg | sudo apt-key add -
+```
+
+成功：
+
+![2.png](https://guoqi-test-1307026204.file.myqcloud.com/202108/2.png)
+
+- 步骤三：添加 OpenResty 官方 APT 仓库。对于 x86_64 或 amd64 系统，可以使用下面的命令：
+
+```shell
+echo "deb http://openresty.org/package/ubuntu $(lsb_release -sc) main" \
+    | sudo tee /etc/apt/sources.list.d/openresty.list
+```
+
+- 而对于 arm64 或 aarch64 系统，则可以使用下面的命令:（我在 M1 上运行的是该命令，上个命令会报错）
+
+```shell
+echo "deb http://openresty.org/package/arm64/ubuntu $(lsb_release -sc) main" \
+    | sudo tee /etc/apt/sources.list.d/openresty.list
+```
+
+- 步骤四：更新 APT 索引：
+
+```shell
+sudo apt-get update
+```
+
+然后就可以像下面这样安装软件包，比如 `OpenResty`：
+
+```shell
+sudo apt-get -y install openresty
+```
+
+- 步骤五：（可选）可以通过下面命令删除该包和对应的关联包：
+
+```shell
+sudo apt-get -y install --no-install-recommends software-properties-common
+```
+
+成功安装 `OpenResty` :
+
+![3.png](https://guoqi-test-1307026204.file.myqcloud.com/202108/3.png)
+
+3. 重新运行安装依赖脚本 （参考上面 1. ）
+4. 接着运行 `LUAROCKS_SERVER=https://luarocks.cn` 命令安装依赖：
+
+![4.png](https://guoqi-test-1307026204.file.myqcloud.com/202108/4.png)
+
+运行下面命令即可：
+
+```shell
+curl https://raw.githubusercontent.com/apache/apisix/master/utils/linux-install-luarocks.sh -sL | bash -
+```
+
+![5.png](https://guoqi-test-1307026204.file.myqcloud.com/202108/5.png)
+
+又出现错误提示了。。。接着运行下面命令：
+
+```shell
+sudo apt install wget sudo unzip
+```
+
+接着我们重新运行一次：
+
+```shell
+curl https://raw.githubusercontent.com/apache/apisix/master/utils/linux-install-luarocks.sh -sL | bash -
+```
+
+然后我们继续运行安装依赖的命令：`LUAROCKS_SERVER=https://luarocks.cn make deps`
+终于成功了，哭了，坑确实太多了。
```

Comment:
I would suggest backporting your fix or patches or methods to the official docs :)